### PR TITLE
gh-118908: Protect the REPL subprocess with a timeout in tests

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-118908

### CODE REVIEW

File: Lib/test/test_pyrepl/test_pyrepl.py [hunk 1]
```diff
@@ -885,5 +886,10 @@ def run_repl(self, repl_input: str | list[str], env: dict | None = None) -> tupl
 
         os.close(master_fd)
         os.close(slave_fd)
-        exit_code = process.wait()
+        try:
+            exit_code = process.wait(timeout=SHORT_TIMEOUT)
+        except subprocess.TimeoutExpired:
+            process.terminate()
```
Comment (vstinner) [inline Lib/test/test_pyrepl/test_pyrepl.py]: Calling kill() is enough.

**Review by vstinner — COMMENTED** [review]
LGTM

File: Lib/test/test_pyrepl/test_pyrepl.py (hunk 1 above)
Comment (pablogsal) [reply Lib/test/test_pyrepl/test_pyrepl.py]: But I want to double kill

File: Lib/test/test_pyrepl/test_pyrepl.py [hunk 2]
```diff
@@ -885,5 +886,10 @@ def run_repl(self, repl_input: str | list[str], env: dict | None = None) -> tupl
...
+            process.kill()
```
Comment (pablogsal) [inline Lib/test/test_pyrepl/test_pyrepl.py]:
```suggestion
            process.kill()
```

File: Lib/test/test_pyrepl/test_pyrepl.py (hunk 1 above)
Comment (vstinner) [reply Lib/test/test_pyrepl/test_pyrepl.py]: It's up to you.

Comment (vstinner) [reply Lib/test/test_pyrepl/test_pyrepl.py]: ☠️☠️

**Review by lysnikolaou — APPROVED** [review]
LGTM!

**Pull request merged**